src/main/java/com/mitchej123/hodgepodge/mixins/late/bibliocraft/MixinBibliocraftPatchPacketExploits.java

```diff
@@ -68,6 +88,34 @@ public class MixinBibliocraftPatchPacketExploits {
         }
     }
 
+    @ModifyVariable(
+            method = "handleFancySignUpdate",
+            at = @At(
+                    value = "INVOKE_ASSIGN",
+                    target = "Lnet/minecraft/world/World;getTileEntity(III)Lnet/minecraft/tileentity/TileEntity;"))
+    private TileEntity hodgepodge$handleFancySignUpdate(TileEntity tile) {
+        if (!(tile instanceof TileEntityFancySign)) {
+            return null;
+        }
+        EntityPlayerMP player = ((NetHandlerPlayServer) currentEvent.handler).playerEntity;
+        PlayerInteractEvent pie = new PlayerInteractEvent(
+                player,
+                PlayerInteractEvent.Action.RIGHT_CLICK_BLOCK,
+                tile.xCoord,
+                tile.yCoord,
+                tile.zCoord,
+                0,
+                tile.getWorldObj());
+        if (MinecraftForge.EVENT_BUS.post(pie)) {
+            // Event cancelled by a world protection plugin or otherwise
+            // Don't kick as there are legitimate ways this could happen (open a gui, then someone claims the chunk,
+            // then exit gui)
+            Common.log.warn(player.getDisplayName() + " tried to edit a sign without permission");
```

Comment:
can it be made so we have uniform logs about cheats by using this `MarkerManager.getMarker("SuspiciousPackets")` ?

Reply:
Preferably also log the GameProfile instead. getDisplayName can be influenced by plugins that offer nickname functionality